main.go

```diff
@@ -16,102 +16,51 @@ package main
 
 import (
 	"context"
-	"fmt"
-	"net/http"
 	"os"
+	"os/signal"
+	"path/filepath"
 	"strings"
+	"syscall"
 
-	"github.com/gorilla/mux"
-	"github.com/spf13/pflag"
-	cli "github.com/virtual-kubelet/node-cli"
-	"github.com/virtual-kubelet/node-cli/opts"
-	"github.com/virtual-kubelet/node-cli/provider"
-	"github.com/virtual-kubelet/systemk/pkg/system"
-	"github.com/virtual-kubelet/systemk/systemd"
-	"k8s.io/klog/v2"
+	"github.com/pkg/errors"
+	"github.com/virtual-kubelet/systemk/cmd"
+	"github.com/virtual-kubelet/systemk/internal/provider"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	"github.com/virtual-kubelet/virtual-kubelet/log/klogv2"
 )
 
 var (
 	buildVersion = "N/A"
 	buildTime    = "N/A"
-	k8sVersion   = "v1.18.4" // Inject this build time by parsing mod.go
+	k8sVersion   = "v1.18.15" // This should follow the version of k8s.io/kubernetes we are importing
```

Comment:
I think this can be removed as we don't wont to be importaint k8s.io/kubernetes

Reply:
Well, users will need to know if systemk is compatible w/ their Kubernetes cluster. For instance, this won't work w/ a Kubernetes <1.20 as soon as you import `k8s.io/kubelet v0.20.2` for providing Pod metrics.

I think we should abide to https://kubernetes.io/docs/setup/release/version-skew-policy/#kubelet.

Reply:
Ah if by "importaint" you didn't mean important but importing, the comment is misleading but my comment above stands.

Reply:
We should be careful by providing those features that are so tied to a k8s version esp if they are so new. Why not focus first on the plain podspec integration and worry about this later when we actually want and need this?

Reply:
This is about the kubelet API not the core API. Pod Stats is a kubelet API.